Update README for NFD hard dependency drop (#127)
openshift/special-resource-operator: README.md

```

# Node Feature Discovery recipe dependency
One challenge when trying to configure a cluster with a special resource is to determine which nodes have a special resource and which do not.
SRO is only aware of the underlying OS and kernel version, which are enough to build (optionally) driver containers and schedule them based on kernel affinity.

> The scheduling process takes place using labels in the nodes, therefore recipes typically need nodes these to be in place.

Reviewer: K8s docs uses ["attaching labels to X"](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#step-one-attach-label-to-the-node) - I feel like the "in the nodes" doesn't fit here. How about: `using labels on the nodes` or `using nodes' labels`.

> recipes typically need nodes these to be in place

Missed words?

Author: Reworked the sentence, let me know what you think

Reviewer: Thanks, I think it's better

```

# Node Feature Discovery recipe dependency
There is a general problem when trying to configure a cluster with a special resource. One does not know which nodes have a special resource and which do not. SRO is aware of the underlying OS and kernel versions only, which are enough to build (optionally) driver containers and schedule them based on kernel affinity. The schedule process takes place using labels in the nodes, therefore recipes typically need nodes these to be in place.

>  To address this, recipes rely on the [NFD operator](https://github.com/openshift/cluster-nfd-operator). NFD will label the host with node specific attributes, like PCI cards, kernel or OS version and more. The .yaml template files in a special resource recipe can use these NFD labels in their nodeSelector fields to ensure that the software stack is run only on the nodes with the hardware feature. See [upstream NFD](https://github.com/kubernetes-sigs/node-feature-discovery) for more info.

Reviewer 2: ```suggestion
To address this, recipes often rely on the [NFD operator](https://github.com/openshift/cluster-nfd-operator).
NFD labels the host with node-specific attributes such as PCI cards, kernel or OS version, and more.
The `.yaml` template files in a special resource recipe can use these NFD labels in their `nodeSelector` fields to ensure that the software stack is run only on the nodes with the hardware feature.
See [upstream NFD](https://github.com/kubernetes-sigs/node-feature-discovery) for more information.
```
